pkg/plugins/jwe/api.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package jwe
+
+import "github.com/google/tink/go/keyset"
+
+// Provider interface for Crypto ctx
+type Provider interface {
+}
+
+// Creator method to create new pluggable JWE service
+type Creator func(prov Provider) (AriesJWE, error)
+
+// AriesJWE is an interface defining a pluggable JWE builder operations provider to Aries agents.
+// It's responsible for building JWE structures for either 1: sealing them or 2: open serialized ones. It depends on
+// on the AriesCrypto to encrypt/decrypt content and AriesKMS to get key handles needed for crypto operations.
+type AriesJWE interface {
+	// Seal will encrypt and wrap msg for the sender key represented by skh and the recipients keys represented in rkh
+	//  into a JWE envelope then serializes it
+	// returns:
+	//		[]byte base64UrlEncoding of the serialized JWE envelope
+	//		error in case of an error
+	Seal(msg []byte, skh *keyset.Handle, rkh []*keyset.Handle) ([]byte, error)
```

Comment:
Where is the Seal and Open terminology coming from?

Reply:
it's for a JWE `envelope` which is sealed by encryption and serialization and then opened by deserialization and decryption.